src/ucs/datastruct/queue.h

```diff
@@ -107,22 +116,22 @@ static inline void ucs_queue_del_iter(ucs_queue_head_t *queue, ucs_queue_iter_t
 {
     ucs_assert((iter != NULL) && (*iter != NULL));
 
-    if (queue->ptail == &(*iter)->next) {
+    if (ucs_queue_is_last(queue, *iter)) {
         queue->ptail = iter; /* deleting the last element */
```

Comment:
comment is really confusing: last could mean 'only one element in queue' and 'no next element in queue' :)

Reply:
maybe ucs_queue_is_back() or ucs_queue_is_tail()?

Reply:
`ucs_queue_is_tail()` sounds good to me

Reply:
yep, tail resolves all misunderstandings :)